driver-sync/src/main/com/mongodb/client/internal/CommandMarker.java

```diff
@@ -41,52 +41,63 @@
     private MongoClient client;
     private final ProcessBuilder processBuilder;
 
-    CommandMarker(final Map<String, Object> options) {
-        String connectionString;
-
-        if (options.containsKey("mongocryptdURI")) {
-            connectionString = (String) options.get("mongocryptdURI");
+    CommandMarker(final boolean isBypassAutoEncryption, final Map<String, Object> options) {
+        if (isBypassAutoEncryption) {
+            processBuilder = null;
+            client = null;
         } else {
-            connectionString = "mongodb://localhost:27020";
-        }
+            if (!options.containsKey("mongocryptdBypassSpawn") || !((Boolean) options.get("mongocryptdBypassSpawn"))) {
+                processBuilder = new ProcessBuilder(createMongocryptdSpawnArgs(options));
+                startProcess();
+            } else {
+                processBuilder = null;
+            }
 
-        if (!options.containsKey("mongocryptdBypassSpawn") || !((Boolean) options.get("mongocryptdBypassSpawn"))) {
-            processBuilder = new ProcessBuilder(createMongocryptdSpawnArgs(options));
-            startProcess();
-        } else {
-            processBuilder = null;
-        }
+            String connectionString;
 
-        client = MongoClients.create(MongoClientSettings.builder()
-                .applyConnectionString(new ConnectionString(connectionString))
-                .applyToClusterSettings(new Block<ClusterSettings.Builder>() {
-                    @Override
-                    public void apply(final ClusterSettings.Builder builder) {
-                        builder.serverSelectionTimeout(1, TimeUnit.SECONDS);
-                    }
-                })
-                .build());
+            if (options.containsKey("mongocryptdURI")) {
+                connectionString = (String) options.get("mongocryptdURI");
+            } else {
+                connectionString = "mongodb://localhost:27020";
+            }
+
+            client = MongoClients.create(MongoClientSettings.builder()
+                    .applyConnectionString(new ConnectionString(connectionString))
+                    .applyToClusterSettings(new Block<ClusterSettings.Builder>() {
+                        @Override
+                        public void apply(final ClusterSettings.Builder builder) {
+                            builder.serverSelectionTimeout(1, TimeUnit.SECONDS);
+                        }
+                    })
+                    .build());
+        }
     }
 
     RawBsonDocument mark(final String databaseName, final RawBsonDocument command) {
-        try {
+        if (client != null) {
```

Comment:
The driver should never try to mark a command in this case, and silently not marking could cause the driver to send unencrypted data to the server.

This should be an IllegalStateException instead.